<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>128</width>
    <height>128</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>LuxMark</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/images/resources/SLG_luxball_sppm_small.png</normaloff>:/images/resources/SLG_luxball_sppm_small.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>160</width>
     <height>120</height>
    </size>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="MainVerticalSplitter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QSplitter" name="MainHorizontalSplitter">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QGraphicsView" name="RenderView"/>
       <widget class="QWidget" name="layoutWidget">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="hardwareDevicesLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
           </property>
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Hardware Devices&lt;/p&gt;&lt;p&gt;(For selected devices only modes)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::NoTextInteraction</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTreeView" name="HardwareView">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="frameShape">
            <enum>QFrame::Box</enum>
           </property>
           <property name="animated">
            <bool>true</bool>
           </property>
           <attribute name="headerVisible">
            <bool>false</bool>
           </attribute>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QTextEdit" name="LogView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <widget class="QMenu" name="menu_Mode">
    <property name="title">
     <string>&amp;Mode</string>
    </property>
    <addaction name="action_CUDA_GPUs"/>
    <addaction name="action_OpenCL_GPUs"/>
    <addaction name="action_Native"/>
    <addaction name="action_Custom"/>
    <addaction name="separator"/>
    <addaction name="action_StressTest_CUDA_GPUs"/>
    <addaction name="action_StressTest_OpenCL_GPUs"/>
    <addaction name="action_StressTest_Native"/>
    <addaction name="action_StressTest_Custom"/>
    <addaction name="separator"/>
    <addaction name="action_LuxCoreUI"/>
    <addaction name="separator"/>
    <addaction name="action_Pause"/>
   </widget>
   <widget class="QMenu" name="menu_Scene">
    <property name="title">
     <string>&amp;Scene</string>
    </property>
    <addaction name="action_WallPaper"/>
    <addaction name="action_HallBench"/>
    <addaction name="action_Food"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Mode"/>
   <addaction name="menu_Scene"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="action_CUDA_GPUs">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Benchmark (CUDA GPUs)</string>
   </property>
  </action>
  <action name="action_Food">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Food (Brute force path tracing)</string>
   </property>
   <property name="toolTip">
    <string>Food (Brute force path tracing)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="action_OpenCL_GPUs">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Benchmark (OpenCL GPUs)</string>
   </property>
  </action>
  <action name="action_Pause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Pause</string>
   </property>
  </action>
  <action name="action_Native">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Benchmark (Native C++)</string>
   </property>
  </action>
  <action name="action_HallBench">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Hall Bench (Path tracing + Global Illumination cache)</string>
   </property>
   <property name="toolTip">
    <string>Hall Bench (Path tracing + Global Illumination cache)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="action_Custom">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Benchmark (Selected devices only)</string>
   </property>
  </action>
  <action name="action_WallPaper">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Wall Paper (Brute force bidirectional path tracing)</string>
   </property>
   <property name="toolTip">
    <string>Wall Paper (Brute force bidirectional path tracing)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_StressTest_CUDA_GPUs">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Stress Test (CUDA GPUs)</string>
   </property>
  </action>
  <action name="action_StressTest_OpenCL_GPUs">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Stress Test (OpenCL GPUs)</string>
   </property>
  </action>
  <action name="action_StressTest_Native">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Stress Test (Native C++)</string>
   </property>
  </action>
  <action name="action_LuxCoreUI">
   <property name="text">
    <string>LuxCoreUI (CUDA or OpenCL GPUs)</string>
   </property>
  </action>
  <action name="action_StressTest_Custom">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Stress Test (Selected devices only)</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>exitApp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_About</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Food</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setFoodScene()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_OpenCL_GPUs</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_BENCHMARK_OCL_GPU()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_CUDA_GPUs</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_BENCHMARK_CUDA_GPU()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Pause</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_PAUSE()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Native</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_BENCHMARK_NATIVE()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_HallBench</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setHallBenchScene()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Custom</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_BENCHMARK_CUSTOM()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_WallPaper</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setWallPaperScene()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_StressTest_CUDA_GPUs</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_STRESSTEST_CUDA_GPU()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_StressTest_OpenCL_GPUs</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_STRESSTEST_OCL_GPU()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_StressTest_Native</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_STRESSTEST_NATIVE()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_LuxCoreUI</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_DEMO_LUXCOREUI()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_StressTest_Custom</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode_STRESSTEST_CUSTOM()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>exitApp()</slot>
  <slot>showAbout()</slot>
  <slot>setFoodScene()</slot>
  <slot>setHallBenchScene()</slot>
  <slot>setMode_PAUSE()</slot>
  <slot>setWallPaperScene()</slot>
  <slot>setMode_BENCHMARK_CUDA_GPU()</slot>
  <slot>setMode_BENCHMARK_OCL_GPU()</slot>
  <slot>setMode_BENCHMARK_NATIVE()</slot>
  <slot>setMode_BENCHMARK_CUSTOM()</slot>
  <slot>setMode_STRESSTEST_CUDA_GPU()</slot>
  <slot>setMode_STRESSTEST_OCL_GPU()</slot>
  <slot>setMode_STRESSTEST_NATIVE()</slot>
  <slot>setMode_STRESSTEST_CUSTOM()</slot>
  <slot>setMode_DEMO_LUXCOREUI()</slot>
 </slots>
</ui>
